app/Http/Requests/PostRequest.php

```diff
@@ -26,7 +26,7 @@ public function rules()
         }
 
         return [
-            'campaign_id' => 'required|integer',
+            'campaign_id' => 'required_without:action_id|integer',
             'northstar_id' => 'nullable|objectid',
             'type' => 'required|string|in:photo,voter-reg,text,share-social',
             // @TODO: eventually, deprecate action in the payload and make action_id required when all systems have been updated.
```

Comment:
just below this should `action_id` also be `required_without:campaign_id`?

Reply:
great catch!